modules.go

```diff
@@ -22,7 +22,7 @@ func (ca *Cagent) initModules() {
 			return storcli.CreateModule(ca.Config.StorCLI.BinaryPath)
 		},
 		func() monitoring.Module {
-			return raid.CreateModule()
+			return raid.CreateModule(ca.Config.SoftwareRAIDMonitoring)
```

Comment:
Why should it add the module in case it is not enabled?

Reply:
The problem is different modules have different behavior of enabling/disabling them. I think it's good idea to encapsulate this behavior inside the module.
For example,  storcli module is enabled based on having non-empty "binary path" config value.

Reply:
Ah, I see. Thanks for the explanation!